src/script/utils/icons.ts

```diff
@@ -83,6 +83,26 @@ export function findBestAppIcon(icons: Icon[] | null | undefined): Icon | null {
   );
 }
 
+/**
+ * Check if an image succesfully loads.
+ * @param icon_url The url to the icon to test.
+ * @returns An boolean value based on whether the image loads or nothing if the image never
+ * fires the load or error event.
+ */
+export function checkImageUrl(icon_url: string): boolean | void {
+  const image = new Image();
+
+  image.onload = () => {
+    return true;
+  };
+  image.onerror = () => {
+    return false;
+  };
+
+  image.src = icon_url;
+}
+
+
```

Comment:
Does this actually work? Doesn't it always return void?  Seems we should be using a promise here, setting its result in the `onload` and `onerror` functions, then returning that. Am I missing something?

Reply:
Well, it was tested and it returns the boolean value correctly. It should never run into a case where it returns void really. On it being a promise, I guess it could although it works fine sync too. Maybe I should just improve the return type here?

Reply:
Hmmm. It's possible I'm missing something. But returning a value from the lambda won't affect the checkImageUrl.

Here's a test fiddle: https://jsfiddle.net/j20g13v7/